core/execution/market.go

```diff
@@ -721,7 +726,23 @@ func (m *Market) OnTick(ctx context.Context, t time.Time) bool {
 		}
 	}
 
-	// check auction, if any
+	// MTM if enough time has elapsed, we are not in auction, and we have a non-zero mark price.
+	// we MTM in leaveAuction before deploying LP orders like we did before, but we do update nextMTM there
+	if mp := m.getCurrentMarkPrice(); mp != nil && !mp.IsZero() && !m.nextMTM.After(t) && !m.as.InAuction() {
+		m.nextMTM = t.Add(m.mtmDelta) // add delta here
```

Comment:
Super nitpicky/paranoid, but I think when exiting the auction we first do MTM and then update `m.nextMTM` wouldn't it be safer to have the same order of operations in both cases in cases there's some future change that makes `confirmMTM` somehow dependent on `m.nextMTM`?

Reply:
It shouldn't really matter. The reason why I set the nextMTM at the end when leaving auction is because I wanted to change as little as possible around the `leaveAuction` flow. It's kind of a self-contained thing, and setting the MTM time is kind of the same as `m.as.Left()`  (ie confirming we're done)

Reply:
Cool, fair enough 👍